src/libpriv/rpmostree-postprocess.c

```diff
@@ -2144,7 +2145,12 @@ rpmostree_commit (int            rootfs_fd,
         goto out;
     }
   
-  if (refname)
+  if (refname && write_commitid_to)
```

Comment:
Because there is more code tied to the existence of refname like setting a parent revision.